src/lib.rs

```diff
@@ -3,6 +3,22 @@
 //! DBC files are descriptions of CAN frames.
 //! See [this post](https://www.kvaser.com/developer-blog/an-introduction-j1939-and-dbc-files/)
 //! for an introduction.
+//!
+//! # Usage
+//!
+//! ```
+//! use dbc_codegen::{codegen, Config, FeatureConfig};
+//!
+//! let config = Config::builder()
+//!     .dbc_name("foo.dbc")
+//!     .dbc_content(include_bytes!("../testing/dbc-examples/example.dbc"))
```

Comment:
bit confusing to use both this and the file one in the same example, maybe just copy-pasta into two examples that show different stuff?

Reply:
Ah that was my original API. maybe follow up with something like this

```rust
impl Config {
    pub fn build_file(path: impl AsRef<Path>) -> Self {
        let path = path.as_ref();
        let dbc_file_name = path
            .file_name()
            .unwrap_or_else(|| path)
            .to_string_lossy();

        Config::builder()
            .dbc_name(dbc_file_name) // ah shit it's borrowed. well make it a cow then!
            .dbc_content(std::fs::read_to_string(&path))
    }
}
```